solver/src/single_order_solver.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::Result;
+use futures::future;
+use rand::prelude::SliceRandom;
+
+use crate::{
+    liquidity::{LimitOrder, Liquidity},
+    settlement::Settlement,
+    solver::Solver,
+};
+
+#[async_trait::async_trait]
+/// Implementations of this trait know how to settle a single limit order (not taking advantage of batching multiple orders together)
+pub trait SingleOrderSolving {
+    /// Return a settlement for the given limit order (if possible)
+    async fn settle_order(&self, order: LimitOrder) -> Result<Settlement>;
+
+    fn name(&self) -> &'static str;
+}
+
+/// Maximum number of sell orders to consider for settlements.
+///
+/// This is mostly out of concern to avoid rate limiting and because
+/// requests may a non-trivial amount of time.
+const MAX_SETTLEMENTS: usize = 5;
+
+pub struct SingleOrderSolver {
```

Comment:
ubernit: I think in this case, it makes sense to make this generic on the single order solving type.